spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -1349,8 +1377,6 @@ trait OptimisticTransactionImpl extends TransactionalWrite
             removeFilesHistogram.foreach(_.insert(r.getFileSize))
           case _: SetTransaction =>
             numSetTransaction += 1
-          case m: Metadata =>
-            assertMetadata(m)
```

Comment:
Just to make sure that I for this right: we don't need this anymore here as `updateMetadataAndProtocolWithRequiredFeatures` already takes care of this check?

Reply:
Yes, there has been no change in the number of times assertMetadata is called.